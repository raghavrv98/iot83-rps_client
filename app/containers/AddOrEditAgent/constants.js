/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
