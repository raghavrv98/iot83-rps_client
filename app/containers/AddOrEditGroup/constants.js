/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
