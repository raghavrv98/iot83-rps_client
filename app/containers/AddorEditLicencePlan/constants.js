/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";