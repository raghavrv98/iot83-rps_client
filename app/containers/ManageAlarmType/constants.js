/*
 *
 * ManageAlarmType constants
 *
 */

export const DEFAULT_ACTION = "app/ManageAlarmType/DEFAULT_ACTION";

export const GET_ALARM_TYPE = "app/ManageAlarmType/GET_ALARM_TYPE";
export const GET_ALARM_TYPE_SUCCESS = "app/ManageAlarmType/GET_ALARM_TYPE_SUCCESS";
export const GET_ALARM_TYPE_FAILURE = "app/ManageAlarmType/GET_ALARM_TYPE_FAILURE";

export const GET_ALARM_CATEGORY = "app/ManageAlarmType/GET_ALARM_CATEGORY";
export const GET_ALARM_CATEGORY_SUCCESS = "app/ManageAlarmType/GET_ALARM_CATEGORY_SUCCESS";
export const GET_ALARM_CATEGORY_FAILURE = "app/ManageAlarmType/GET_ALARM_CATEGORY_FAILURE";

export const ADD_ALARM_TYPE = "app/ManageAlarmType/ADD_ALARM_TYPE";
export const ADD_ALARM_TYPE_SUCCESS = "app/ManageAlarmType/ADD_ALARM_TYPE_SUCCESS";
export const ADD_ALARM_TYPE_FAILURE = "app/ManageAlarmType/ADD_ALARM_TYPE_FAILURE";

//ADD_ALARM_TYPE, ADD_ALARM_TYPE_SUCCESS, ADD_ALARM_TYPE_FAILURE, GET_ALARM_TYPE, GET_ALARM_TYPE_SUCCESS, GET_ALARM_TYPE_FAILURE
