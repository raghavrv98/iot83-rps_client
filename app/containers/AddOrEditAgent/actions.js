/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
