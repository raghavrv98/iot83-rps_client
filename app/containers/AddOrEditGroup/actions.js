/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
