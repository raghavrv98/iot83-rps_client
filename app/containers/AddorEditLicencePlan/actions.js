/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}