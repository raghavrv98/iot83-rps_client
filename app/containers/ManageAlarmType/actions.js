/*
 *
 * ManageAlarmType actions
 *
 */

import {ADD_ALARM_TYPE, GET_ALARM_TYPE, GET_ALARM_CATEGORY} from './constants'

export function getAlarmType() {
  return {
    type: GET_ALARM_TYPE
  };
}

export function getAlarmCategory() {
  return {
    type: GET_ALARM_CATEGORY
  };
}


export function addAlarmType(payload) {
  return {
    type: ADD_ALARM_TYPE,
    payload
  };
}

