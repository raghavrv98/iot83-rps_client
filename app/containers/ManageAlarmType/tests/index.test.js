import React, { Component } from 'react';
import { shallow } from 'enzyme';
import history from "utils/history";
import { getAlarmType, getAlarmCategory } from "../actions";
import { GET_ALARM_TYPE, ADD_ALARM_TYPE, GET_ALARM_CATEGORY } from "../constants";
import { ManageAlarmType, mapDispatchToProps } from "../index";
const alarmList = [
  {
    _id: "9bed0d5ca79011e9863a0242ac110016",
    createdAt: 1563257514466,
    name: "edfedeefdf",
    other: "fvcerdfedfv",
    overallGeometry: "dfe3dfedc",
    plantId: "f29b79c09f0911e986810242ac110011",
    processFluid: "efe3dcfed",
    updatedAt: 1563257547048
  }
];

describe('< ManageAlarmType />', () => {

  it('should test component when render', () => {
    spyOn(ManageAlarmType.prototype, 'render').and.callThrough();
    const wrapper = shallow(<ManageAlarmType getAlarmType={getAlarmType} getAlarmCategory={getAlarmCategory} list={alarmList} />);
    expect(wrapper).toBeDefined();
    expect(ManageAlarmType.prototype.render).toHaveBeenCalledTimes(1);
  })

  it('should test component when recieve props', () => {
    spyOn(ManageAlarmType.prototype, 'componentWillReceiveProps').and.callThrough();
    const wrapper = shallow(<ManageAlarmType
      getAlarmType={getAlarmType}
      getAlarmCategory={getAlarmCategory}
      list={undefined}
      listError={undefined}
      deleteSuccess={undefined}
      deleteFailure={undefined}
      getAlarmsCategorySuccess={undefined}
      getAlarmsCategoryFailure={undefined}
    />);
    expect(wrapper).toBeDefined();
    wrapper.setState({ alarmList });
    wrapper.setProps({
      list: alarmList,
      listError: "No Error Found.",
      deleteSuccess: "bb157d6738bb4886b77fd25635301010",
      deleteFailure: "No Error Found in delete.",
      getAlarmsCategorySuccess: [],
      getAlarmsCategoryFailure: "No Error Found in alarm category."
    });
    expect(ManageAlarmType.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
  })

  it('match snapshots', () => {
    const wrap = shallow(<ManageAlarmType getAlarmType={getAlarmType} getAlarmCategory={getAlarmCategory} />)
    expect(wrap).toMatchSnapshot()
  })

  it("Function testing on mapDispatchToProps", () => {
    const mockParams = {
      params: {
        payload: "demo",
      }
    };
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).getAlarmType();
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: GET_ALARM_TYPE
    });

    mapDispatchToProps(dispatch).addAlarmType(mockParams.params.payload);
    expect(dispatch.mock.calls[1][0]).toEqual({
      type: ADD_ALARM_TYPE,
      payload: "demo"
    });

    mapDispatchToProps(dispatch).getAlarmCategory();
    expect(dispatch.mock.calls[2][0]).toEqual({
      type: GET_ALARM_CATEGORY
    });

  });

  it("test componentWillMount", async () => {
    const match = {
      params: {
        types: "demo"
      }
    };
    const fetchAlarmList = jest.fn();
    spyOn(ManageAlarmType.prototype, "componentWillMount").and.callThrough();
    shallow(
      <ManageAlarmType
        match={match}
        getAlarmType={getAlarmType}
        getAlarmCategory={getAlarmCategory}
      />
    );
    expect(ManageAlarmType.prototype.componentWillMount).toHaveBeenCalledTimes(1);
    expect(fetchAlarmList).toHaveBeenCalledTimes(0);
  });
});