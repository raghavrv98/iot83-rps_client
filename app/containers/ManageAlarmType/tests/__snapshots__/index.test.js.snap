// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< ManageAlarmType /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      ManageAlarmType
    </title>
    <meta
      content="Description of ManageAlarmType"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p>
        <FormattedMessage
          defaultMessage="Manage Alarms"
          id="app.common.messages.manageAlarms"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="All Available Alarm Type"
          id="app.containers.ManageAlarmType.allAvailableAlarmType"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </h5>
    </div>
    <div
      className="flex-item fx-b30"
    />
  </div>
  <div
    className="contentForm addAlarm"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        <form
          className="flex"
          onSubmit={[Function]}
        >
          <div
            className="fx-b50"
          >
            <div
              className="createAlarm"
            >
              <p
                className="categoryErrorMessage"
              />
              <p
                className="categoryErrorMessage"
              />
              <p
                className="categoryErrorMessage"
              />
              <div
                className="dropdown selectAlarmCategory"
              >
                <button
                  className="btn dropdown-toggle"
                  data-toggle="dropdown"
                  type="button"
                >
                  <span
                    className="typeIcon"
                  >
                    <img
                      src="IMAGE_MOCK"
                    />
                  </span>
                  <FormattedMessage
                    defaultMessage="Select Category"
                    id="app.containers.ManageAlarmType.selectCategory"
                    values={Object {}}
                  >
                    <Component />
                  </FormattedMessage>
                </button>
                <ul
                  className="dropdown-menu"
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  placeholder="Enter Alarm Name"
                />
              </div>
              <button
                className="btn btn-transparent"
                type="submit"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-check"
                />
              </button>
            </div>
          </div>
          <div
            className="fx-b50"
          />
        </form>
      </div>
      <SkeletonLoader
        skeleton="skeletonAlarmType"
      />
    </div>
    <div
      className="form-group justify-content-end mt-3"
    >
      <button
        className="btn btn-danger"
        id="saveAlarmType"
        onClick={[Function]}
        type="button"
      >
        <i
          className="far fa-check-circle"
        />
        <FormattedMessage
          defaultMessage="Save"
          id="app.common.messages.save"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </button>
    </div>
  </div>
</div>
`;
